File: extractor-core/src/main/java/ru/mydesignstudio/database/metadata/extractor/extractors/model/ColumnModel.java
package ru.mydesignstudio.database.metadata.extractor.extractors.model;

import lombok.Data;
import lombok.EqualsAndHashCode;
import org.apache.commons.lang.StringUtils;

@Data
@EqualsAndHashCode(of = "columnId")
public class ColumnModel {
  private String schemaName;
  private String tableName;
  private String columnName;
  private int columnId;
  private String dataType;
  private String dataTypeExt;
  private int dataLength;
  private Integer dataPrecision;
  private Integer dataScale;
  private String nullable;
  private String defaultDefinition;
  private String primaryKey;
  private String foreign_key;
  private String uniqueKey;
  private String checkConstraint;
  private String comments;

  public boolean isNullable() {
    return StringUtils.equals("Y", getNullable());
  }
}

File: extractor-core/src/main/kotlin/ru/mydesignstudio/database/metadata/extractor/extract/result/ColumnModel.kt
package ru.mydesignstudio.database.metadata.extractor.extract.result

import org.apache.commons.lang.StringUtils

data class ColumnModel(
        val schemaName: String,
        val tableName: String,
        val columnName: String,
        val columnId: Int,
        val dataType: String,
        val dataTypeExt: String,
        val dataLength: Int,
        val dataPrecision: Int,
        val dataScale: Int,
        val nullable: String,
        val defaultDefinition: String,
        val primaryKey: String,
        val foreignKey: String,
        val uniqueKey: String,
        val checkConstraint: String,
        val comments: String
) {
    // TODO, looks incorrect, needs to be checked
    fun isNullable(): Boolean {
        return StringUtils.equalsIgnoreCase("Y", nullable)
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as ColumnModel

        if (columnId != other.columnId) return false

        return true
    }

    override fun hashCode(): Int {
        return columnId
    }


}